arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/LowGasStipendPrecompileCallTests.java

```diff
@@ -0,0 +1,260 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.precompiles;
+
+import static net.consensys.linea.zktracer.instructionprocessing.callTests.Utilities.populateMemory;
+import static net.consensys.linea.zktracer.module.constants.GlobalConstants.WORD_SIZE;
+import static org.hyperledger.besu.datatypes.Address.ALTBN128_ADD;
+import static org.hyperledger.besu.datatypes.Address.ALTBN128_MUL;
+import static org.hyperledger.besu.datatypes.Address.ALTBN128_PAIRING;
+import static org.hyperledger.besu.datatypes.Address.BLAKE2B_F_COMPRESSION;
+import static org.hyperledger.besu.datatypes.Address.ECREC;
+import static org.hyperledger.besu.datatypes.Address.ID;
+import static org.hyperledger.besu.datatypes.Address.MODEXP;
+import static org.hyperledger.besu.datatypes.Address.RIPEMD160;
+import static org.hyperledger.besu.datatypes.Address.SHA256;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.zktracer.module.hub.Hub;
+import net.consensys.linea.zktracer.module.oob.OobOperation;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.hyperledger.besu.datatypes.Address;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class LowGasStipendPrecompileCallTests {
+
+  // Enums for the different testing scenarios
+  enum ValueParameter {
+    ZERO,
+    NON_ZERO;
+
+    boolean isZeroArgument() {
+      return this == ZERO;
+    }
+
+    boolean isNonZeroArgument() {
+      return this == NON_ZERO;
+    }
+  }
+
+  enum GasParameter {
+    ZERO,
+    ONE,
+    COST_MINUS_ONE,
+    COST,
+    COST_PLUS_ONE;
+  }
+
+  @ParameterizedTest
+  @MethodSource("lowGasStipendPrecompileCallTestSource")
+  void lowGasStipendPrecompileCallTest(
+      Address precompileAddress, ValueParameter valueParameter, GasParameter gasParameter) {
+    final BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    // In order to actually trigger the insufficient we need to:
+    // - Set a specific args size for BLAKE2F and EC_PAIRING
+    // - Set the r value of BLAKE2F to have precompileCost > gasBonus
+    // - Populate the memory with a large enough number of words for SHA256, RIPEMD160, and ID
+    //   to have precompileCost > gasBonus.
+    final int value = valueParameter.isZeroArgument() ? 0 : 1;
+    final int argsSize; // depends on the called precompile
+    final int argsOffset = valueParameter.isZeroArgument() ? 0 : 1;
+    final int retSize = valueParameter.isZeroArgument() ? 0 : 1;
```

Comment:
I would suggest setting `retSize` to something nonzero always:
```rust
ECRECOVER, SHA2, RIPEMD, ECPAIRING -> WORD_SIZE
ECADD, ECMUL -> 2 * WORD_SIZE
BLAKE2f -> 2 * WORD_SIZE

Reply:
Also note that what you call `retSize` / `retOffset` / `argSize` / `argOffset` we typically call `rac`, `rao`, `cds`, `cdo`.